src/components/users/subscriptions/ko/runtime/subscriptions.ts

```diff
@@ -116,8 +118,8 @@ export class Subscriptions {
         subscription.isSRegenerating(true);
 
         try {
-            const isDelegationEnabled = await this.isDelegationEnabled(subscriptionId);
-            if (isDelegationEnabled) {
+            const isDelegationApplied = await this.applyDelegation(subscriptionId);
```

Comment:
Since you have the redirect inside `applyDelegation` method, there is no point to have this `if (isDelegationApplied){}` statement. If we get to this point, it means no delegation happened. Therefore, `applyDelegation` method could return `void` for simplicity.